src/main/java/com/o19s/es/ltr/feature/store/ScriptFeature.java

```diff
@@ -0,0 +1,305 @@
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            LtrScript ltrScript = (LtrScript) o;
+            return Objects.equals(scoreFunction, ltrScript.scoreFunction) &&
+                    Objects.equals(featureSupplier, ltrScript.featureSupplier);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(scoreFunction, featureSupplier);
+        }
+
+        LtrScript(ScriptScoreFunction scoreFunction, FeatureSupplier featureSupplier) {
+            this.scoreFunction = scoreFunction;
+            this.featureSupplier = featureSupplier;
+        }
+
+        @Override
+        public String toString(String field) {
+            return "LtrScript:" + field;
+        }
+
+
+        @Override
+        public Weight createWeight(IndexSearcher searcher, boolean needsScores, float boost) throws IOException {
+            return new LtrScriptWeight(this);
+        }
+
+
+        class LtrScriptWeight extends FeatureVectorWeight {
+
+            protected LtrScriptWeight(Query query) {
+                super(query);
+            }
+
+            @Override
+            public Explanation explain(LeafReaderContext context, LtrRanker.FeatureVector vector, int doc) throws IOException {
+                LtrScript.this.featureSupplier.set(() -> vector);
+                // TODO; figure out why we need to pass an explanation here
+                return scoreFunction.getLeafScoreFunction(context).explainScore(doc, Explanation.match(1F, "TODO"));
+            }
+
+            @Override
+            public Scorer scorer(LeafReaderContext context, Supplier<LtrRanker.FeatureVector> vectorSupplier) throws IOException {
+                LtrScript.this.featureSupplier.set(vectorSupplier);
+                LeafScoreFunction leafScoreFunction = scoreFunction.getLeafScoreFunction(context);
+                DocIdSetIterator iterator = DocIdSetIterator.all(context.reader().maxDoc());
+                return new Scorer(this) {
+                    @Override
+                    public int docID() {
+                        return iterator.docID();
+                    }
+
+                    @Override
+                    public float score() throws IOException {
+                        return (float) leafScoreFunction.score(iterator.docID(), 0F);
+                    }
+
+                    @Override
+                    public DocIdSetIterator iterator() {
+                        return iterator;
+                    }
+                };
+            }
+
+            @Override
+            public void extractTerms(Set<Term> terms) {
+            }
+
+            @Override
+            public boolean isCacheable(LeafReaderContext ctx) {
+                return false;
+            }
+        }
+    }
+
+
+    static class FeatureSupplier extends AbstractMap<String, Float> implements Supplier<LtrRanker.FeatureVector> {
```

Comment:
perhaps we should extract this class into its own file.
Having a dedicated unit test for it would be nice I think.

Reply:
fixed, still need to add unit tests though.